Assets/HoloToolkit/Input/Scripts/InputManager.cs

```diff
@@ -228,54 +235,62 @@ public void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<
             }
 
             // Use focused object when OverrideFocusedObject is null.
-            GameObject focusedObject = (OverrideFocusedObject == null) ? GazeManager.Instance.HitObject : OverrideFocusedObject;
+            GameObject focusedObject = OverrideFocusedObject ?? GazeManager.Instance.HitObject;
 
-            // Send the event to global listeners
+            // Send the event to global listeners.
             for (int i = 0; i < globalListeners.Count; i++)
             {
-                // Global listeners should only get events on themselves, as opposed to their hierarchy
+                // Global listeners should only get events on themselves, as opposed to their hierarchy.
                 ExecuteEvents.Execute(globalListeners[i], eventData, eventHandler);
             }
 
-            // Handle modal input if one exists
+            // Check if the event has already been used by the global listeners.
+            if (eventData.used)
```

Comment:
There's an assumption that global listeners should never use events. I guess this code is correct, but maybe we should instead show an error or something like that if a global listener uses an event?